src/Core/Worker.php

```diff
@@ -445,42 +445,68 @@ public static function execute(array $queue): bool
 	}
 
 	/**
-	 * Execute a function from the queue
+	 * Slow the execution down if the system load is too high
 	 *
-	 * @param array   $queue       Workerqueue entry
-	 * @param string  $funcname    name of the function
-	 * @param array   $argv        Array of values to be passed to the function
-	 * @param boolean $method_call boolean
 	 * @return void
-	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	private static function execFunction(array $queue, string $funcname, array $argv, bool $method_call)
+	public static function coolDown()
 	{
-		$a = DI::app();
+		$load_cooldown      = DI::config()->get('system', 'worker_load_cooldown');
+		$processes_cooldown = DI::config()->get('system', 'worker_processes_cooldown');
 
-		$cooldown = DI::config()->get('system', 'worker_cooldown', 0);
-		if ($cooldown > 0) {
-			Logger::debug('Pre execution cooldown.', ['cooldown' => $cooldown, 'id' => $queue['id'], 'priority' => $queue['priority'], 'command' => $queue['command']]);
-			sleep($cooldown);
+		if (($load_cooldown == 0) && ($processes_cooldown == 0)) {
+			return;
 		}
 
-		$load_cooldown      = DI::config()->get('system', 'worker_load_cooldown');
-		$processes_cooldown = DI::config()->get('system', 'worker_processes_cooldown');
+		$sleeping = false;
 
-		while ((($load_cooldown > 0) || ($processes_cooldown > 0)) && ($load = System::getLoadAvg())) {
+		while ($load = System::getLoadAvg()) {
```

Comment:
I guess the root cause for high CPU loads are such loops:
````
<?php
while (true) {
// Do something
}
````
Now just run that little script and you will see that one CPU core is 100% busy with it. If you just add a `usleep(100);` at the end of the loop (still inside of it!) then you will see that the CPU load goes back drastically. Comment it out again and the CPU load goes up again.

Reply:
Please have a look at the loop. Later in that loop there is a `sleep(1)` - and at the end is a `break;`.

Reply:
Okay, I just saw the diff-view here. :-)